src/basics.md

```diff
@@ -411,6 +412,45 @@ fn run() -> Result<()> {
 # quick_main!(run);
 ```
 
+[error-handling]: #error-handling
+<a name="error-handling"></a>
+## Handle errors correctly in main
+
+[![error-chain-badge]][error-chain] [![cat-rust-patterns-badge]][cat-rust-patterns]
+
+Uses [error-chain] to handle error occurred when trying to open a file that does not exist.
+
+First, we call `error_chain!{}` inside a new module named  _errors_. This will create the default types `Error`, `ErrorKind`, `ResultExt` and ` Result`, and store them in `errors`. To be able to use these types we must call `use errors::*`.
+
+To handle the erros that may occur when calling `open_foo()`, it must have return type `Reuslt<()>`. We then use [`if let`] when calling it to detect if an error has occurred, and to handle it in an apropiated way.
+```rust
+#[macro_use]
+extern crate error_chain;
+
+mod errors {
+    error_chain!{}
```

Comment:
@somebrel looks good to me :+1: we can defer showing `chain_err` with custom error and error unrolling to a more advanced example. But its your story (I'm just here to help) I am game either way.

I would only drop the `::` prefix